src/role-access/services/allocate-role.service.ts

```diff
@@ -25,7 +26,11 @@ export class AllocateRoleService {
     }
   }
 
-  public specificAccessApproval(specificAccessStateData: SpecificAccessStateData, period: Period): Observable<any> {
+  public specificAccessApproval(specificAccessStateData: SpecificAccessStateData, dtperiod: Period): Observable<any> {
+    const period = {
+      startDate : this.durationService.setUTCTimezone(dtperiod.startDate),
```

Comment:
There can be case where startDate / endDate is not available. Do you need to check for it?